File: gumroad.py
import json

import requests

LICENSE_VERIFY_ENDPOINT = "https://api.gumroad.com/v2/licenses/verify"


def verify_product_exists(product_permalink):
    response = requests.post(LICENSE_VERIFY_ENDPOINT,
                             data={'product_permalink': product_permalink,
                                   'license_key': "XXXXX"})
    response_json = json.loads(response.content)
    return "message" in response_json and response_json["message"] != "No product exists with that custom or unique permalink."


def verify_license(product_permalink, license_id):
    response = requests.post(LICENSE_VERIFY_ENDPOINT,
                             data={'product_permalink': product_permalink,
                                   'license_key': license_id,
                                   "increment_uses_count": "false"})
    response_json = json.loads(response.content)
    return response_json["success"] and not response_json["purchase"]["refunded"]

File: src/main/java/Utils/GumRoad.java
package Utils;

import com.fasterxml.jackson.databind.ObjectMapper;
import org.apache.http.NameValuePair;
import org.apache.http.client.config.CookieSpecs;
import org.apache.http.client.config.RequestConfig;
import org.apache.http.client.entity.UrlEncodedFormEntity;
import org.apache.http.client.methods.CloseableHttpResponse;
import org.apache.http.client.methods.HttpPost;
import org.apache.http.impl.client.CloseableHttpClient;
import org.apache.http.impl.client.HttpClients;
import org.apache.http.message.BasicNameValuePair;

import java.io.*;
import java.util.ArrayList;
import java.util.List;

/**
 * Main endpoint for interfacing with GumRoad
 */
public class GumRoad {
    public static Boolean GetLicenseValid(String productPermalink, String license)
            throws IOException {
        CloseableHttpClient client = HttpClients.custom()
                .setDefaultRequestConfig(RequestConfig.custom()
                        .setCookieSpec(CookieSpecs.STANDARD).build())
                .build();

        HttpPost post = new HttpPost("https://api.gumroad.com/v2/licenses/verify");

        List<NameValuePair> params = new ArrayList<>();
        params.add(new BasicNameValuePair("product_permalink", productPermalink));
        params.add(new BasicNameValuePair("license_key", license));
        params.add(new BasicNameValuePair("increment_uses_count", "false"));
        post.setEntity(new UrlEncodedFormEntity(params));

        CloseableHttpResponse resp = client.execute(post);
        if (resp.getStatusLine().getStatusCode() != 200) return false;

        BufferedReader rdr = new BufferedReader(new InputStreamReader(resp.getEntity().getContent()));
        StringBuilder resultStr = new StringBuilder();
        String strBuf;
        while ((strBuf = rdr.readLine()) != null)
            resultStr.append(strBuf);

        client.close();

        ObjectMapper mapper = new ObjectMapper();
        GumRoadResponse gumRoadResponse = mapper.readValue(resultStr.toString(), GumRoadResponse.class);

        return gumRoadResponse.success && !gumRoadResponse.purchase.refunded;
    }
}
